src/java/CMakeLists.txt

```diff
@@ -123,7 +123,11 @@ else()
 endif()
 
 # run the build process
-set(GRADLE_ARGS cmakeBuild -DcmakeBuildDir=${JAVA_OUTPUT_DIR} -DnativeLibDir=${JAVA_PACKAGE_LIB_DIR})
+set(GRADLE_ARGS cmakeBuild)
+if(PUBLISH_JAVA_MAVEN)
+  set(GRADLE_ARGS ${GRADLE_ARGS} publishToMavenLocal)
+endif()
+set(GRADLE_ARGS ${GRADLE_ARGS} -DcmakeBuildDir=${JAVA_OUTPUT_DIR} -DnativeLibDir=${JAVA_NATIVE_LIB_DIR})
```

Comment:
```suggestion
if(PUBLISH_JAVA_MAVEN)
  list(APPEND GRADLE_ARGS publishToMavenLocal)
endif()
list(APPEND GRADLE_ARGS -DcmakeBuildDir=${JAVA_OUTPUT_DIR} -DnativeLibDir=${JAVA_NATIVE_LIB_DIR})
```

Reply:
OK, thanks for the review and sorry for the late follow up (I've been on holiday). I'm addressing this in https://github.com/microsoft/onnxruntime-genai/pull/1570 , which is also rebased against main).